clusterman/batch/node_migration.py

```diff
@@ -53,36 +58,97 @@ def configure_initial(self):
         self.run_interval = staticconf.read_int("batches.node_migration.run_interval_seconds", 60)
+        self.event_visibilty_timeout = staticconf.read_int(
+            "batches.node_migration.event_visibilty_timeout_seconds", 15 * 60
+        )
         for pool in get_pool_name_list(self.options.cluster, self.POOL_SCHEDULER):
             load_cluster_pool_config(self.options.cluster, pool, self.POOL_SCHEDULER, None)
             pool_config_namespace = POOL_NAMESPACE.format(pool=pool, scheduler=self.POOL_SCHEDULER)
             pool_config = staticconf.config.get_namespace(pool_config_namespace).get_config_dict()
             if self.POOL_SETTINGS_PARENT in pool_config:
                 self.migration_configs[pool] = pool_config[self.POOL_SETTINGS_PARENT]
                 self.add_watcher({pool: get_pool_config_path(self.options.cluster, pool, self.POOL_SCHEDULER)})
+                if self.migration_configs[pool]["trigger"].get("event_queue", False):
+                    self.pools_accepting_events.add(pool)
         self.logger.info(f"Found node migration configs for pools: {list(self.migration_configs.keys())}")
 
-    def fetch_event_queues(self, queues: Collection[str]):
-        """Tail a collection of SQS queue for migration trigger events
+    def _fetch_all_from_queue(self) -> Generator[dict, None, None]:
+        """Fetch all data from SQS queue
 
-        :param Collection[str] queues: SQS queue URLs
+        :return: yields messages
         """
-        # TODO: everything
+        try:
+            while True:
+                messages = self.sqs_client.receive_message(
+                    QueueUrl=self.event_queue,
+                    MaxNumberOfMessages=self.EVENT_FETCH_BATCH,
+                    VisibilityTimeout=self.event_visibilty_timeout,
+                    WaitTimeSeconds=10,
+                ).get("Messages", [])
+                yield from messages
+                if len(messages) < self.EVENT_FETCH_BATCH:
+                    break
+        except ClientError as e:
+            self.logger.exception(f"Issue retrieving events from {self.event_queue}: {e}")
+
+    def _get_worker_setup(self, pool: str) -> Optional[WorkerSetup]:
+        """Build worker setup for
+
+        :param str pool: name of the pool
+        :return: migration worker setup object
+        """
+        try:
+            if pool in self.migration_configs:
+                return WorkerSetup.from_config(self.migration_configs[pool])
+        except Exception as e:
+            self.logger.exception(f"Bad migration configuration for pool {pool}: {e}")
+        return None
+
+    def fetch_event_queue(self) -> Collection[MigrationEvent]:
+        """Tail a collection of SQS queue for migration trigger events"""
+        events = map(MigrationEvent.from_event, self._fetch_all_from_queue())
+        return set(events) - self.events_in_progress
```

Comment:
it never set. Probably you will have in the next commits. just in case

Reply:
yep, I'll get `events_in_progress` populated once I get to work on the worker-process management code